core/arch/arm/mm/core_mmu.c

```diff
@@ -914,10 +914,11 @@ static void assign_mem_granularity(struct tee_mmap_region *memory_map)
 
 static unsigned int get_va_width(void)
 {
-	if (IS_ENABLED(ARM64))
-		return 64 - __builtin_ctzll(CFG_LPAE_ADDR_SPACE_SIZE);
-	else
-		return 32;
+	if (IS_ENABLED(ARM64) || IS_ENABLED(CFG_WITH_LPAE)) {
```

Comment:
I think `if (IS_ENABLED(CFG_WITH_LPAE))` is enough: Aarch64 implies LPAE.

Reply:
It's only `ARM64` that can have anything different from `32`.

Reply:
Right, will fix

Reply:
> It's only `ARM64` that can have anything different from `32`.

Really? So why would one enable LPAE on arm32?

Reply:
OK so the correct code is just `if (IS_ENABLED(ARM64))`, right? No need to test for `CFG_WITH_LPAE`?

Reply:
Hmmm, should the config assert that `CFG_LPAE_ADDR_SPACE_SIZE=(1 << 32)` when `ARM32` is defined?

Reply:
> > It's only `ARM64` that can have anything different from `32`.
> 
> Really? So why would one enable LPAE on arm32?

This should not be confused with `CFG_CORE_ARM64_PA_BITS`

Reply:
> Hmmm, should the config assert that `CFG_LPAE_ADDR_SPACE_SIZE=(1 << 32)` when `ARM32` **and `CFG_WITH_LPAE` are enabled**?

Sure we can have something for that too.